libs/core/src/lib/message-strip/message-strip.component.ts

```diff
@@ -0,0 +1,194 @@
+        'role': 'alert',
+        '[attr.id]': 'id',
+    },
+    encapsulation: ViewEncapsulation.None,
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class MessageStripComponent implements OnInit, OnChanges, CssClassBuilder {
+
+    /** user's custom classes */
+    private _class: string = '';
+    @Input()
+    set class(userClass: string) {
+        this._class = userClass;
+        this.buildComponentCssClass();
+    }
+
+    /** Whether the message strip is dismissible. */
+    private _dismissible: boolean = true;
+    @Input()
+    set dismissible(dismissible: boolean) {
+        this._dismissible = dismissible;
+        this.buildComponentCssClass();
+    }
+
+    get dismissible(): boolean {
+        return this._dismissible;
+    }
+
+    /** The default message strip does not have an icon. 
+     * The other types (warning, success, information and error) have icons by default. 
+     * To remove the icon set the property to true. */
+    private _noIcon: boolean = false;
+    @Input()
+    set noIcon(noIcon: boolean) {
+        this._noIcon = noIcon;
+        this.buildComponentCssClass();
+    }
+
+    /** The type of the message strip. Can be one of *warning*, *success*, *information*, *error* or null. */
+    private _type: string;
+    @Input()
+    set type(type: string) {
+        this._type = type;
+        this.buildComponentCssClass();
+    }
+
+    /** Id for the message-strip component. If omitted, a unique one is generated. */
+    private _id: string = 'fd-message-strip-' + messageStripUniqueId++;
+    @Input()
+    set id(id: string) {
+        this._id = id;
+        this.buildComponentCssClass();
+    }
+
+    get id(): string {
+        return this._id;
+    }
+
+    /** Id of the element that labels the message-strip. */
+    private _ariaLabelledBy: string = null;
+    @Input()
+    set ariaLabelledBy(ariaLabelledBy: string) {
+        this._ariaLabelledBy = ariaLabelledBy;
+        this.buildComponentCssClass();
+    }
+
+    get ariaLabelledBy(): string {
+        return this._ariaLabelledBy;
+    }
+
+    /** Aria label for the message-strip component element. */
+    private _ariaLabel: string = null;
+    @Input()
+    set ariaLabel(ariaLabel: string) {
+        this._ariaLabel = ariaLabel;
+        this.buildComponentCssClass();
+    }
+
+    get ariaLabel(): string {
+        return this._ariaLabel;
+    }
+
+    /** Aria label for the dismiss button. */
+    private _dismissLabel: string = 'Dismiss';
+    @Input()
+    set dismissLabel(dismissLabel: string) {
+        this._dismissLabel = dismissLabel;
+        this.buildComponentCssClass();
+    }
+
+    get dismissLabel(): string {
+        return this._dismissLabel;
+    }
+
+    /** Width of the message-strip. */
+    private _width: string;
+    @Input()
+    set width(width: string) {
+        this._width = width;
+        this.buildComponentCssClass();
+    }
+
+    get width(): string {
+        return this._width;
```

Comment:
What is the advantage of using private variables/getters and setters for the inputs on this component?

Reply:
This is the approach we agreed on since we introduced 
`@applyCssClass
    buildComponentCssClass(): string {
        return [...].join(' ');
   }`
for composing the classes.